karma.conf.js

```diff
@@ -1,81 +1,50 @@
-// Karma configuration
-// Generated on Fri Dec 05 2014 16:49:29 GMT-0500 (EST)
+const path = require('path');
 
 module.exports = function(config) {
   config.set({
 
-    // base path that will be used to resolve all patterns (eg. files, exclude)
     basePath: '',
-
-
-    // frameworks to use
-    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-    frameworks: ['jspm', 'jasmine'],
-
-    jspm: {
-      // Edit this to your needs
-      loadFiles: ['src/**/*.js', 'test/**/*.js']
-    },
-
-
-    // list of files / patterns to load in the browser
-    files: [],
-
-
-    // list of files to exclude
-    exclude: [
-    ],
-
-
-    // preprocess matching files before serving them to the browser
-    // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+    frameworks: ["jasmine"],
+    files: ["test/*.spec.ts"],
     preprocessors: {
-      'test/**/*.js': ['babel'],
-      'src/**/*.js': ['babel']
+      "**/*.ts": ["webpack"]
     },
-    'babelPreprocessor': {
-      options: {
-        sourceMap: 'inline',
-        presets: [ 'es2015-loose', 'stage-1'],
-        plugins: [
-          'syntax-flow',
-          'transform-decorators-legacy',
-          'transform-flow-strip-types'
+    webpack: {
+      mode: "development",
+      resolve: {
+        extensions: [".ts", ".js"],
+        modules: ["src", "node_modules"],
+        alias: {
+          src: path.resolve(__dirname, "src")
+        }
+      },
+      devtool: "cheap-module-eval-source-map",
+      module: {
+        rules: [
+          {
+            test: /\.ts$/,
+            loader: "ts-loader",
+            exclude: /node_modules/,
+            options: {
+              transpileOnly: true
```

Comment:
I typically turn this on only when a code base is volatile and I just want tests to run even with typing errors. It also speeds up the compilation to have this on.
You might want to remove this or make it parameterized as a measure of robustness for this mature library. It's probably better to see typing errors during testing as well.

Reply:
Fixed, thanks !